src/pkg/term/colorizer.go

```diff
@@ -30,26 +35,95 @@ const (
 	DebugColor = termenv.ANSIBrightBlack // Gray
 )
 
-// DoColor returns true if the provided output's profile is not Ascii.
-func DoColor(o *termenv.Output) bool {
-	return o.Profile != termenv.Ascii
+func NewTerm(stdout, stderr io.Writer) *Term {
+	t := &Term{
+		outw:   stdout,
+		errw:   stderr,
+		stdout: termenv.NewOutput(stdout),
+		stderr: termenv.NewOutput(stderr),
+	}
+	t.hasDarkBg = t.stdout.HasDarkBackground()
+	if hasTermInEnv() {
+		fout, outIsFile := stdout.(interface{ Fd() uintptr })
+		ferr, errIsFile := stderr.(interface{ Fd() uintptr })
+		if outIsFile && errIsFile {
+			t.isTerminal = term.IsTerminal(int(fout.Fd())) && term.IsTerminal(int(ferr.Fd()))
```

Comment:
@edwardrf regression: old code checked `os.Stdin`. This breaks `echo a | defang config set KEY`